server/cmd/api/api/display.go

```diff
@@ -0,0 +1,271 @@
+		xvfbResp, xvfbErr := s.ProcessExec(ctx, oapi.ProcessExecRequestObject{Body: &restartXvfbReq})
+		if xvfbErr != nil {
+			return oapi.SetResolution500JSONResponse{InternalErrorJSONResponse: oapi.InternalErrorJSONResponse{Message: "failed to restart Xvfb"}}, nil
+		}
+
+		// Check if Xvfb restart succeeded
+		if execResp, ok := xvfbResp.(oapi.ProcessExec200JSONResponse); ok {
+			if execResp.ExitCode != nil && *execResp.ExitCode != 0 {
+				return oapi.SetResolution500JSONResponse{InternalErrorJSONResponse: oapi.InternalErrorJSONResponse{Message: "Xvfb restart failed"}}, nil
+			}
+		}
+
+		// Wait for Xvfb to be ready
+		log.Info("waiting for Xvfb to be ready")
+		waitCmd := []string{"-lc", "sleep 2"}
+		waitReq := oapi.ProcessExecRequest{Command: "bash", Args: &waitCmd}
+		s.ProcessExec(ctx, oapi.ProcessExecRequestObject{Body: &waitReq})
+
+		// Restart Chromium
+		log.Info("restarting chromium after Xvfb restart")
+		restartChromeCmd := []string{"-lc", "supervisorctl restart chromium"}
+		restartChromeEnv := map[string]string{}
+		restartChromeReq := oapi.ProcessExecRequest{Command: "bash", Args: &restartChromeCmd, Env: &restartChromeEnv}
+		chromeResp, chromeErr := s.ProcessExec(ctx, oapi.ProcessExecRequestObject{Body: &restartChromeReq})
+		if chromeErr != nil {
+			log.Error("failed to restart chromium after Xvfb restart", "error", chromeErr)
+			// Still return success since Xvfb restart succeeded
+			return oapi.SetResolution200JSONResponse{Ok: true}, nil
+		}
+
+		// Check if Chromium restart succeeded
+		if execResp, ok := chromeResp.(oapi.ProcessExec200JSONResponse); ok {
+			if execResp.ExitCode != nil && *execResp.ExitCode != 0 {
+				log.Error("chromium restart failed", "exit_code", *execResp.ExitCode)
+			} else {
+				log.Info("chromium restarted successfully")
+			}
+		}
+
+		log.Info("Xvfb resolution updated", "display", display, "width", width, "height", height)
+		return oapi.SetResolution200JSONResponse{Ok: true}, nil
+	}
+}
+
+// anyRecordingActive returns true if any registered recorder is currently recording.
+func (s *ApiService) anyRecordingActive(ctx context.Context) bool {
+	for _, r := range s.recordManager.ListActiveRecorders(ctx) {
+		if r.IsRecording(ctx) {
+			return true
+		}
+	}
+	return false
+}
+
+// countEstablishedTCPSessions returns the number of ESTABLISHED TCP connections for the given local port.
+// Implementation shells out to netstat, which is present in the image (net-tools).
+func (s *ApiService) countEstablishedTCPSessions(ctx context.Context, port int) int {
+	cmd := exec.CommandContext(ctx, "/bin/bash", "-lc", fmt.Sprintf("netstat -tn 2>/dev/null | awk '$6==\"ESTABLISHED\" && $4 ~ /:%d$/ {count++} END{print count+0}'", port))
+	out, err := cmd.Output()
+	if err != nil {
+		return 0
+	}
+	val := strings.TrimSpace(string(out))
+	if val == "" {
+		return 0
+	}
+	i, err := strconv.Atoi(val)
+	if err != nil {
+		return 0
+	}
+	return i
+}
```

Comment:
Alternately you could query neko's API since it exposes session information https://github.com/onkernel/neko/blob/aa0487f68ebbff1056d7355ec2da127986e5be5f/server/openapi.yaml#L224-L243